detectable/src/main/java/com/synopsys/integration/detectable/detectables/maven/cli/MavenCliExtractor.java

```diff
@@ -50,17 +50,17 @@ public MavenCliExtractor(DetectableExecutableRunner executableRunner, MavenCodeL
 
     //TODO: Limit 'extractors' to 'execute' and 'read', delegate all other work.
     public Extraction extract(File directory, ExecutableTarget mavenExe, MavenCliExtractorOptions mavenCliExtractorOptions) throws ExecutableFailedException {
-        String[] mavenCommand = mavenCliExtractorOptions.getMavenBuildCommand()
-                                    .map(cmd -> cmd.replace("dependency:tree", ""))
-                                    .map(String::trim)
-                                    .map(cmd -> cmd.split(" "))
-                                    .orElse(new String[] {});
+        List<String> commandArguments = mavenCliExtractorOptions.getMavenBuildCommandArguments()
+                                     .orElse(new LinkedList<>())
+                                     .stream()
+                                     .map(cmd -> cmd.replace("dependency:tree", ""))
```

Comment:
Since the command has already been split, should we just filter out an argument that equals the dependency:tree? .filter(it -> it.equals("dependency:tree")

Reply:
That's a good call